src/txmempool.cpp

```diff
@@ -533,10 +533,17 @@ void CTxMemPool::removeForReorg(CChainState& active_chainstate, int flags)
                 const Coin &coin = active_chainstate.CoinsTip().AccessCoin(txin.prevout);
                 if (m_check_ratio != 0) assert(!coin.IsSpent());
                 unsigned int nMemPoolHeight = active_chainstate.m_chain.Tip()->nHeight + 1;
-                if (coin.IsSpent() || (coin.IsCoinBase() && ((signed long)nMemPoolHeight) - coin.nHeight < COINBASE_MATURITY)) {
-                    txToRemove.insert(it);
-                    break;
-                }             
+                    if (coin.nHeight < Params().GetConsensus().multiAlgoDiffChangeTarget) {
```

Comment:
@Jongjan88 , I think we have a formatting error here as the if statement is aligned to the same level as the subsequent if statement.

I think it should be the following:

```
for (const CTxIn& txin : tx.vin) {
    indexed_transaction_set::const_iterator it2 = mapTx.find(txin.prevout.hash);
    if (it2 != mapTx.end())
        continue;
    const Coin &coin = active_chainstate.CoinsTip().AccessCoin(txin.prevout);
    if (m_check_ratio != 0) assert(!coin.IsSpent());
    unsigned int nMemPoolHeight = active_chainstate.m_chain.Tip()->nHeight + 1;
    if (coin.nHeight < Params().GetConsensus().multiAlgoDiffChangeTarget) {
        if (coin.IsSpent() || (coin.IsCoinBase() && ((signed long)nMemPoolHeight) - coin.nHeight < COINBASE_MATURITY)) { 
            txToRemove.insert(it); 
            break; 
        } 
    } else { 
        if (coin.IsSpent() || (coin.IsCoinBase() && ((signed long)nMemPoolHeight) - coin.nHeight < COINBASE_MATURITY_2)) { 
            txToRemove.insert(it); 
            break; 
        }
    }
}

```

Reply:
Ty. I changed it.